src/main/java/com/amplitude/HttpTransport.java

```diff
@@ -206,6 +210,9 @@ private EventsRetryResult retryEventsOnce(String userId, String deviceId, List<E
       case PAYLOAD_TOO_LARGE:
         shouldRetry = true;
         break;
+      case TIMEOUT:
+          triggerEventCallbacks(events, response.code, response.error);
```

Comment:
Why are we trigger callbacks here? I don't see shouldRetry is updated, so this is still going to retry again right?

Reply:
I wonder in this case we would get more callbacks than expected. The error callbacks are more calling once after either success or we don't retry anymore

Reply:
removed the two line and changed the test for this